bin/grade_item.py

```diff
@@ -504,38 +654,89 @@ def just_grade_item(next_directory,next_to_grade,which_untrusted):
+
+    partial_path = os.path.join(queue_obj["gradeable"],queue_obj["who"],str(queue_obj["version"]))
+    item_name = os.path.join(queue_obj["semester"],queue_obj["course"],"submissions",partial_path)
+    results_path = os.path.join(SUBMITTY_DATA_DIR,"courses",queue_obj["semester"],queue_obj["course"],"results",partial_path)
+
+    # clean out all of the old files if this is a re-run
+    shutil.rmtree(results_path,ignore_errors=True)
+    # create the directory (and the full path if it doesn't already exist)
+    os.makedirs(results_path)
+    # unzip the file & clean up
+    unzip_this_file(my_results_zip_file,results_path)
+    os.remove(my_results_zip_file)
+
+    # add information to the database
     insert_database_version_data.insert_to_database(
-        obj["semester"],
-        obj["course"],
-        obj["gradeable"],
-        obj["user"],
-        obj["team"],
-        obj["who"],
-        True if obj["is_team"] else False,
-        str(obj["version"]))
+        queue_obj["semester"],
+        queue_obj["course"],
+        queue_obj["gradeable"],
+        queue_obj["user"],
+        queue_obj["team"],
+        queue_obj["who"],
+        True if queue_obj["is_team"] else False,
+        str(queue_obj["version"]))
 
-    print ("pid",my_pid,"finished grading ", next_to_grade, " in ", gradingtime, " seconds")
+    submission_path = os.path.join(SUBMITTY_DATA_DIR,"courses",item_name)
 
-    grade_items_logging.log_message(is_batch_job,which_untrusted,submission_path,"grade:",gradingtime,grade_result)
+    is_batch_job = queue_obj["is_batch_job"]
+    gradingtime=queue_obj["gradingtime"]
+    grade_result=queue_obj["grade_result"]
 
-    with open(os.path.join(tmp_logs,"overall.txt"),'a') as f:
-        f.write("FINISHED GRADING!")
+    print ("pid",os.getpid(),"finished grading ", item_name, " in ", int(gradingtime), " seconds")
 
-    # save the logs!
-    shutil.copytree(tmp_logs,os.path.join(results_path,"logs"))
+    grade_items_logging.log_message(is_batch_job,"unzip",item_name,"grade:",'{0:.3f}'.format(gradingtime),grade_result)
 
 
-    # --------------------------------------------------------------------
-    # REMOVE TEMP DIRECTORY
-    shutil.rmtree(tmp)
 
+# ==================================================================================
+# ==================================================================================
+def just_grade_item(next_directory,next_to_grade,which_untrusted):
+    # verify the hwcron user is running this script
+    if not int(os.getuid()) == int(HWCRON_UID):
+        grade_items_logging.log_message("ERROR: must be run by hwcron")
+        raise SystemExit("ERROR: the grade_item.py script must be run by the hwcron user")
+
+    autograding_zip,submission_zip = prepare_autograding_and_submission_zip(next_directory,next_to_grade)
+    results_zip = grade_from_zip(autograding_zip,submission_zip,which_untrusted)
+    unpack_grading_results_zip(results_zip)
```

Comment:
I'd like to see us doing a better job of cleaning up after jobs that fail during the process. Right now, any of the commands that raise exceptions (of which there's a bunch) in the middle will leave us with some amount of leftover stuff (and when we move to docker, a leftover container). 

I'd at least like to see a try/finally around `grade_from_zip` and `unpack_grading_results_zip` where the finally calls a cleanup routine. If it's important that we maintain the logs from this failed grading effort for analysis, I'd add a catch clause that copies those files over to somewhere that makes sense and then re-raise the caught exception.